src/sql/src/parsers/create_parser.rs

```diff
@@ -62,17 +62,21 @@ impl<'a> ParserContext<'a> {
         let _ = self.parser.next_token();
         self.parser
             .expect_keyword(Keyword::TABLE)
-            .context(error::SyntaxSnafu)?;
+            .context(SyntaxSnafu)?;
         let if_not_exists =
             self.parser
                 .parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
         let table_name = self.intern_parse_table_name()?;
         let (columns, constraints) = self.parse_columns()?;
+        if !columns.is_empty() {
```

Comment:
It looks like we should check this unconditionally. Empty table is not allowed (and btw is not covered in tests)

Reply:
Interesting empty table is supported for external table:

```sql
CREATE EXTERNAL TABLE IF NOT EXISTS city WITH (location='/var/data/city.csv',format='csv');
```

.. and we have some tests cover this.

Reply:
See `expr_factory::create_external_expr` for an execution time validation.

Reply:
That's a bit different. Constraints are more released on external files. Like a `TIME INDEX` is not required.

Reply:
This condition is added later. Previously I check the constraints unconditionally, and it cause the valid SQL above to fail.

Note that this line is for parsing create_external_table statement.